src/librustc/ty/layout.rs

```diff
@@ -673,12 +673,8 @@ impl<'a, 'gcx, 'tcx> Struct {
     }
 
     pub fn offset_of_field(&self, index: usize) -> Size {
-        assert!(index < self.offset_after_field.len());
-        if index == 0 {
-            Size::from_bytes(0)
-        } else {
-            self.offset_after_field[index-1]
-        }
+        assert!(index < self.offsets.len());
+        self.offsets[index]
     }
```

Comment:
Only reason for this method was the oddity of `offset_after_field`, direct access to `offsets` would be better.